pkg/cluster/install.go

```diff
@@ -49,7 +49,13 @@ func (m *manager) AdminUpdate(ctx context.Context) error {
 
 func (m *manager) Update(ctx context.Context) error {
 	steps := []steps.Step{
+		steps.Action(m.initializeKubernetesClients), // must be first
 		steps.AuthorizationRefreshingAction(m.fpAuthorizer, steps.Action(m.validateResources)),
+		// credentials rotation flow steps
+		steps.Action(m.createOrUpdateClusterServicePrincipalRBAC),
+		steps.Action(m.createOrUpdateDenyAssignment),
+		steps.Action(m.updateAROSecret),
+		steps.Action(m.updateOpenShiftSecret),
```

Comment:
We are doing a simplification here: we are setting up the Azure RBAC, then updating the cluster.  If the principal changes, that means there's a broken period (the alternative is add new principal, update cluster, wait, remove old principal).  Is the expense of the alternative sufficiently higher to warrant the simplification?

The more foundational stuff we do the right way, the easier it is for everyone else to build on, both us and customers.

Reply:
I think I'm going to at least partially answer my own question.

I guess the difficulty here is tracking that all the users of the old SP have gone away, and I guess there is a sliding scale here.

Before merging this PR, we should definitely make sure that these steps are enough to kick all the processes that we're aware of that use the SP (cloud controller, machine API, dns, any others?).  We want to avoid alerts later.

Ideally there would be a way to wait for these processes to restart, but I can imagine that there's not the infrastructure in place to do this in a supportable way.  In which case, let's do it this way.

Reply:
This is still on my radar, but it is time consuming to test :/